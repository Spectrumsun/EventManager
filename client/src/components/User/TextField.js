import React from 'react';

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
      //required
    />
  </div>
);


export default TextField;
