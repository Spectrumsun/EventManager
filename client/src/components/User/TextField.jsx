import React from 'react';
import PropTypes from 'prop-types'; 

const TextField = ({
  type, value, onChange, name, label, placeholder
}) => (
  <div className="form-group">
    <label htmlFor="2">{label}</label>
    <input
      type={type}
      value={value}
      onChange={onChange}
      name={name}
      className="form-control form-control-lg"
      placeholder={placeholder}
    
    />
  </div>
);


TextField.propTypes = {
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  placeholder: PropTypes.string.isRequired
};

export default TextField;
