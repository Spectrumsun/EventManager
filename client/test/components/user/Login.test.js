import React from 'react';
import { mount, shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import render from 'react-test-renderer';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import sinon from 'sinon';
import ConnectedLogin,
{ Login } from '../../../src/components/User/Login';
import TextField from '../../../src/components/User/TextField';


const middleware = [thunk];
const mockStore = configureStore(middleware);
const initialState = {
  isAuthenticated: false,
  user: {},
};
const store = mockStore(initialState);

const props = {
  initUserLogin: sinon.spy(() => new Promise((cb) => {
    cb();
  })),
  history: createMemoryHistory()
};

const mountedWrapper = mount(
  <Provider store={store}>
  <BrowserRouter>
    <ConnectedLogin {...props} />
  </BrowserRouter>
  </Provider>
);

const shallowWrapper = shallow(<Login {...props} />);

const state = {
  email: 'bot@yahoo.com',
  Password: '12345678',
};

const event = {
  preventDefault: jest.fn(),
  target: {
    email: 'bot@yahoo.com',
    password: '12345678',
  }
};

// let wrapper;

describe('<Login /> Component', () => {
  it('should render the <Login />', () => {
    shallow(<Login />);
  });

  it('should render the <Login /> without crashing', () => {
    expect(mountedWrapper).toBeDefined();
    expect(mountedWrapper.find('Login').length).toBe(1);
  });

  it('should match component snapshot', () => {
    const tree = render.create(
      <BrowserRouter >
        <Login {...props} />
      </BrowserRouter>);
    expect(tree).toMatchSnapshot();
  });


  it('should render initial layout of HomePage', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.getElements()).toMatchSnapshot();
  });
  
  it('should have three image on layout', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.find('div').length).toEqual(7);
  });

  it('should have form ', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.find('form').length).toEqual(1);
  });

  it('should have <TextField /> when the page loads', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper.find(TextField)).toHaveLength(2);
  });

  it('calls onChange event', () => {
    sinon.spy(shallowWrapper.instance(), 'onChange');
    shallowWrapper.instance().onChange(event);
    expect(shallowWrapper.instance().onChange.calledOnce).toEqual(true);
  });

  it('calls onSubmit event', () => {
    sinon.spy(shallowWrapper.instance(), 'onSubmit');
    shallowWrapper.setState(state);
    shallowWrapper.instance().onSubmit(event);
    expect(shallowWrapper.instance().onSubmit.calledOnce).toEqual(true);
  });
});

