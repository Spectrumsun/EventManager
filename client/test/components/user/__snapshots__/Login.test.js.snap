// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> Component should match component snapshot 1`] = `
<div>
  <div
    className="container"
    style={
      Object {
        "paddingTop": "150px",
      }
    }
  >
    <h3
      className="center"
    >
      Event Manager
    </h3>
    <div
      className="card loginCard"
      style={
        Object {
          "width": "30rem",
        }
      }
    >
      <div
        className="card-header"
      >
        <h3>
          Login
        </h3>
      </div>
      <div
        className="card-body"
      >
        <div
          className="cont card-body"
        >
          <form
            className="centerform"
            onSubmit={[Function]}
          >
            <div
              className="form-group"
            >
              <label
                htmlFor="2"
              >
                Email
              </label>
              <input
                className="form-control form-control-lg"
                name="email"
                onChange={[Function]}
                placeholder="Vaild Email"
                required={true}
                type="email"
                value=""
              />
            </div>
            <div
              className="form-group"
            >
              <label
                htmlFor="2"
              >
                Password
              </label>
              <input
                className="form-control form-control-lg"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <small>
              <a
                className="center-item"
                href="/forgotpassword"
                onClick={[Function]}
              >
                Forgot password?
              </a>
            </small>
            <div
              className="text-center"
            >
              <button
                className="btn btn-outline-dark"
                disabled={false}
                id="login"
                type="submit"
              >
                Submit
              </button>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Login /> Component should render initial layout of HomePage 1`] = `
Array [
  <div>
    <div
      className="container"
      style={
        Object {
          "paddingTop": "150px",
        }
      }
    >
      <h3
        className="center"
      >
        Event Manager
      </h3>
      <div
        className="card loginCard"
        style={
          Object {
            "width": "30rem",
          }
        }
      >
        <div
          className="card-header"
        >
          <h3>
            Login
          </h3>
        </div>
        <div
          className="card-body"
        >
          <div
            className="cont card-body"
          >
            <form
              className="centerform"
              onSubmit={[Function]}
            >
              <TextField
                label="Email"
                name="email"
                onChange={[Function]}
                placeholder="Vaild Email"
                type="email"
                value=""
              />
              <TextField
                label="Password"
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
              <small>
                <Link
                  className="center-item"
                  replace={false}
                  to="/forgotpassword"
                >
                  Forgot password?
                </Link>
              </small>
              <div
                className="text-center"
              >
                <button
                  className="btn btn-outline-dark"
                  disabled={false}
                  id="login"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
